mvrx/src/main/kotlin/com/airbnb/mvrx/lifecycleAwareLazy.kt

```diff
@@ -26,12 +29,16 @@ class lifecycleAwareLazy<out T>(private val owner: LifecycleOwner, initializer:
     private val lock = this
 
     init {
-        owner.lifecycle.addObserver(object : DefaultLifecycleObserver {
-            override fun onCreate(owner: LifecycleOwner) {
-                if (!isInitialized()) value
-                owner.lifecycle.removeObserver(this)
-            }
-        })
+        // owner.lifecycle.addObserver must be invoked on main thread otherwise addObserver will throw an IllegalStateException.
+        // createUnsafe disables the main thread check.
+        synchronized(lock) {
```

Comment:
I don't think this synchronized is helping here. Nothing can access this in the constructor.

Also, because this class owns the registry, I think it's just fine to add createUnsafe and call it a day.

Reply:
No worries. Updated.